### PR TITLE
Promote `boot_disk_kms_key` of `google_container_node_pool` to GA

### PR DESCRIPTION
**Pull request by upodroid**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/10650

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: promoted `node_config.0.boot_disk_kms_key` of `google_container_node_pool` to GA
```

### CODE REVIEW

File: mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb [hunk 1]
```diff
@@ -1015,21 +1015,21 @@ func TestAccContainerCluster_withSandboxConfig(t *testing.T) {
 		},
 	})
 }
+<% end -%>
 
 func TestAccContainerCluster_withBootDiskKmsKey(t *testing.T) {
-	// KMS key has calculated next rotation time
-	skipIfVcr(t)
```
Comment (melinath) [inline mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb:1021]: Is this no longer necessary? What changed? Or was it unnecessary in the first place?

Comment (melinath) [reply mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb:1021]: It looks like this is obviated by `BootstrapKMSKeyInLocation`.

**Review by melinath — APPROVED** [review]
LGTM assuming the tests pass.

**Comment by melinath** [comment]
GA test run of the new test: https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloud_ProviderGoogleCloudMmUpstream/229918

**Comment by melinath** [comment]
Is there a test that exercises this feature for google_container_node_pool as well that needs to be promoted?

File: mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb (hunk 1 above)
Comment (upodroid) [reply mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb:1021]: Yeah, the test is more efficient now by reusing bootstrap kms keys and making VCR usable

**Comment by upodroid** [comment]
> Is there a test that exercises this feature for google_container_node_pool as well that needs to be promoted?

A separate one isn't needed as the node_config.go logic is used in the cluster resource and all the fields are ForceNew anyway

**Pull request merged**